### PR TITLE
Implement compgen command

### PR DESCRIPTION
**Pull request by suganyasuven**
## 
Purpose
> $title

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/21074

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/Command.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+package io.ballerina.cli.cmd;
+
+import java.util.List;
+
+/**
+ * Command Object.
+ *
+ * @since 2.0.0
+ */
+public class Command {
```
[the remark this answers is not in the document]
Comment (suganyasuven) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/Command.java:27]: This class is used to map the JSON object

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/CompgenCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Test(description = "Suggest flags in accordance with the args")
+    public void testCompgenFlagSuggestion() throws IOException {
+        CompgenCommand compgenCommand = new CompgenCommand(printStream);
+        new CommandLine(compgenCommand).parse("--", "build", "--c");
+        compgenCommand.execute();
+
+        String output = readOutput(true);
+        Assert.assertTrue(output.contains("compile"));
```
[the remark this answers is not in the document]
Comment (suganyasuven) [reply cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/CompgenCommandTest.java:65]: Yes, Checked both(L66).

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/Command.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (hevayo) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/Command.java:27]: Can we use an obvious name ie CommandCompletions

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/CompgenCommand.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        this.outStream = outStream;
+    }
+
+    @CommandLine.Parameters(description = "Command name")
+    private List<String> bashDistCommands;
+
+    @Override
+    public void execute() {
+        List<String> subCommands;
+        List<String> dirContents = getDirContents(this.userDir);
```
Comment (hevayo) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/CompgenCommand.java:65]: Can we load the dir content only if we need them loading it at the top will be a performance hit

**Review by hevayo — APPROVED** [review]

**Review by hevayo — APPROVED** [review]

**Pull request merged**